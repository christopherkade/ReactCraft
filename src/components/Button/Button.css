.rc-button {
  font-size: 1rem;
  background-color: #710504;
  color: #FFC700;
  padding: 0.25rem 2.5rem;
  border-radius: 6px;
  border: 2px ridge #8C8C8C;
  -webkit-text-stroke-width: 0.5px;
  -webkit-text-stroke-color: black;
  text-shadow: 1px 1.5px 1px black;
  box-shadow: inset 0 0 2px 0.5px black;
}

.rc-button:hover {
  cursor: pointer;
}

.rc-button:focus {
  outline: transparent;
}
