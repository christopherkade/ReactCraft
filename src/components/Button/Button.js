import React from "react"
import PropTypes from 'prop-types';

import "./Button.css"

const Button = ({ children, onClick }) => {
  return (
    <button onClick={onClick} className="rc-button">
      {children}
    </button>
  )
}

Button.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  onClick: PropTypes.func,
};

Button.defaultProps = {
  children: 'Enter World',
  onClick: () => { },
};

export default Button
