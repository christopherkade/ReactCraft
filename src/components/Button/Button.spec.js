import React from 'react';
import Button from './Button';
import { render, fireEvent } from 'react-testing-library';

describe('<Button />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Button />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('children prop', () => {
    it('should display Enter World as Button children', () => {
      const { container } = render(<Button>Enter World</Button>);
      expect(container.querySelector('button').textContent).toBe('Enter World');
    });
  });

  describe('onClick prop', () => {
    it('should call onClick function when Button is clicked', () => {
      const onClickMock = jest.fn();
      const { getByText } = render(<Button onClick={onClickMock}>Click me!</Button>);
      fireEvent.click(getByText('Click me!'));
      expect(onClickMock).toHaveBeenCalled();
    });
  });
});
