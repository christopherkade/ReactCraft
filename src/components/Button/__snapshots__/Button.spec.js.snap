// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Snapshots should match snapshot 1`] = `
<div>
  <button
    class="rc-button"
  >
    Enter World
  </button>
</div>
`;
